src/click/types.py

```diff
@@ -663,21 +664,38 @@ class BoolParamType(ParamType):
 
     def convert(
         self, value: t.Any, param: Parameter | None, ctx: Context | None
-    ) -> t.Any:
-        if value in {False, True}:
-            return bool(value)
-
-        norm = value.strip().lower()
-
-        if norm in {"1", "true", "t", "yes", "y", "on"}:
-            return True
+    ) -> bool:
+        if isinstance(value, bool):
+            return value
 
-        if norm in {"0", "false", "f", "no", "n", "off"}:
-            return False
+        # import pdb
+
+        # pdb.set_trace()
+
+        bool_states: dict[str, bool] = RawConfigParser.BOOLEAN_STATES | {
```

Comment:
It seems that unpacking `RawConfigParser.BOOLEAN_STATES` with `**` or using `t.cast()` could fix the typing issue

Reply:
Ah yes, that's good alternatives. I'm not opinionated about which is the preferred way here.